server/src/main/java/io/crate/planner/optimizer/rule/RewriteJoinPlan.java

```diff
@@ -46,6 +46,13 @@ public Pattern<JoinPlan> pattern() {
         return pattern;
     }
 
+    @Override
+    public boolean mandatory() {
+        // this rule is currently mandatory to convert JoinPlan to NestedLoop/HashJoin
+        // and apply further optimizations
```

Comment:
Does this mean we can drop the `build` fallback in the JoinPlan?

Reply:
Yes good point.